tools/wptrunner/wptrunner/environment.py

```diff
@@ -50,6 +51,74 @@ class TestEnvironmentError(Exception):
     pass
 
 
+class _QueueLogger(StructuredLogger):
+    """Structured logger that has a unique name and puts messages on a queue"""
+    def __init__(self, queue):
+        name = str(uuid.uuid4())
+        self.queue = queue
+        super().__init__(name)
+        assert not self.handlers
+
+    def _handle_log(self, data):
+        self.queue.put(data)
+
+
+class LoggerProxy:
+    """Logger frontend that puts messages on a provided queue.
+
+    This is intended for use in wptserve which doesn't directly depend
+    on mozlog, so it presents an API like the stdlib logger module.
+    However to allow the data to be used directly by mozlog, we construct
+    an internal StructuredLogger that is just used to transform the data
+    into the correct format. This is given a unique name to avoid clashing
+    with any other configured loggers.
+    """
+
+    def __init__(self, name, queue):
+        self.name = name
+        self._logger = _QueueLogger(queue)
+
+    def critical(self, *args):
+        self._logger.critical(*args)
+
+    def error(self, *args):
+        self._logger.error(*args)
+
+    def warning(self, *args):
+        self._logger.warning(*args)
+
+    def info(self, *args):
+        self._logger.info(*args)
+
+    def debug(self, *args):
+        self._logger.debug(*args)
+
+
+class ServerLogging:
```

Comment:
Aside from the hard-coded strings, there's nothing really server-specific about this. Consider renaming.